src/algebra/algebra/operations.lean

```diff
@@ -42,19 +42,19 @@ variables (S T : set A) {M N P Q : submodule R A} {m n : A}
 
 /-- `1 : submodule R A` is the submodule R of A. -/
 instance : has_one (submodule R A) :=
-⟨submodule.map (of_id R A).to_linear_map (⊤ : submodule R R)⟩
+⟨submodule.map (algebra.linear_map R A) (⊤ : submodule R R)⟩
 
 theorem one_eq_map_top :
-  (1 : submodule R A) = submodule.map (of_id R A).to_linear_map (⊤ : submodule R R) := rfl
+  (1 : submodule R A) = submodule.map (algebra.linear_map R A) (⊤ : submodule R R) := rfl
+
+@[simp] lemma mem_one {x : A} : x ∈ (1 : submodule R A) ↔ ∃ y, algebra_map R A y = x :=
+by simp only [one_eq_map_top, mem_map, mem_top, true_and, algebra.linear_map_apply]
```

Comment:
I actually wanted this lemma for something before, and never got around to PRing it.

Reply:
Do you think either of these lemmas would be useful?

https://github.com/pygae/lean-ga/blob/9f2f36dd9236faa4403f933e8800c395cd9b2516/src/for_mathlib/algebra/algebra_operations.lean#L15-L24

Reply:
Both would be useful I think. Shall I add them to this PR?

Reply:
(Wait, why don't we just directly define `1` as `algebra.linear_map.range`, instead of this messing about with `map \top`?)

Reply:
I've no opposition to it being defined that way, if you want to try it